Source/WebCore/dom/Element.h

```diff
@@ -409,10 +409,13 @@ class Element : public ContainerNode {
 
     inline ShadowRoot* shadowRoot() const; // Defined in ElementRareData.h
     RefPtr<ShadowRoot> shadowRootForBindings(JSC::JSGlobalObject&) const;
+    RefPtr<Element> deepShadowRootReferenceTargetOrSelf();
+    RefPtr<const Element> deepShadowRootReferenceTargetOrSelf() const;
```

Comment:
Oh, we typically just addd `RefPtr<Element> deepShadowRootReferenceTargetOrSelf() const;`

Reply:
So even if it's returning `this`, it should cast away constness?